server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -121,31 +124,45 @@ private GetMailboxesResponse getMailboxesResponse(GetMailboxesRequest mailboxesR
             .orElseGet(Throwing.supplier(() -> retrieveAllMailboxes(mailboxSession)).sneakyThrow());
     }
 
+    private MailboxFactory.MailboxBuilder getBuilder(
+        MailboxSession mailboxSession,
+        MailboxId mailboxId,
+        Optional<List<MailboxMetaData>> preloadedMailboxMetadata,
+        Optional<Quotas> defaultQuotas) {
+        final MailboxFactory.MailboxBuilder builder = mailboxFactory.builder()
+            .id(mailboxId)
+            .session(mailboxSession);
+        final MailboxFactory.MailboxBuilder builderWithQuotas = defaultQuotas.map(builder::usingPreloadedUserDefaultQuotas).orElse(builder);
+        return preloadedMailboxMetadata.map(builderWithQuotas::usingPreloadedMailboxesMetadata).orElse(builderWithQuotas);
```

Comment:
how about replacing .orElse() by lazily .orElseGet()

Reply:
As the builder is already instantiated, i don't see the need to use the lazy method

Reply:
add CR please